tests/testthat/test-seq_linter.R

```diff
@@ -5,6 +5,28 @@ test_that("other : expressions are fine", {
   expect_lint("function(x) { 1:(length(x) || 1) }", NULL, linter)
 })
 
+test_that("seq_len(...) or seq_along(...) expressions are fine", {
+  linter <- seq_linter()
+  expect_lint("function(x) { seq_len(x) }", NULL, linter)
```

Comment:
could you add tests for other usages of seq() that shouldn't lint? e.g. seq(2, length(x)) and seq(length(x), 2) (since 2:length(x) doesn't lint)

Reply:
Damn, that lints:

``` r
library(lintr)

lint(
  text = "function(x) { seq(2, length(x)) }",
  linters = seq_linter()
)
#> <text>:1:15: warning: [seq_linter] seq(2) is likely to be wrong in the empty edge case. Use seq_len(...) instead.
#> function(x) { seq(2, length(x)) }
#>               ^~~~~~~~~~~~~~~~~
```

<sup>Created on 2022-07-25 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

I think the xpath I am using is incorrect. 

It shouldn't be 

`seq()` + anything in between + bad_funcs

But rather

`seq()` + nothing in between + bad_funcs

Reply:
Bingo!

``` r
library(lintr)

lint(
  text = "function(x) { seq(2, length(x)) }",
  linters = seq_linter()
)

lint(
  text = "function(x) { seq(length(x), 2) }",
  linters = seq_linter()
)
```

<sup>Created on 2022-07-26 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>